frontend/src/scenes/pipeline/pipelineNodeLogic.tsx

```diff
@@ -152,12 +212,87 @@ export const pipelineNodeLogic = kea<pipelineNodeLogicType>([
             ],
         ],
         nodeBackend: [
-            (_, p) => [p.id],
-            (id): PipelineBackend => (typeof id === 'string' ? PipelineBackend.BatchExport : PipelineBackend.Plugin),
+            (s, p) => [s.node, p.id, s.newConfigurationServiceOrPluginID],
+            (node, id, newConfigurationServiceOrPluginID): PipelineBackend | null => {
+                if (node) {
+                    return node.backend
+                }
+                if (id === 'new') {
+                    if (newConfigurationServiceOrPluginID === null) {
+                        return null
+                    } else if (typeof newConfigurationServiceOrPluginID === 'string') {
+                        return PipelineBackend.BatchExport
+                    }
+                    return PipelineBackend.Plugin
+                }
+                if (typeof id === 'string') {
+                    return PipelineBackend.BatchExport
+                }
+                return PipelineBackend.Plugin
+            },
+        ],
+        maybeNodePlugin: [
```

Comment:
This is the only thing that's a bit smelly 😄 We previously had `maybePlugin` and `maybeBatchExport` but I refactored that to a common `Node` type, so that it's as unimportant as possible what the destinations is _powered_ by. In this case we're coming back to that a bit and I think a better abstraction might be a `Node | NewNode` type for `node`, which would only have `backend` + info directly derived from the plugin config (all things known before the destination is saved).

Reply:
I tried the `newNode` route initially, but then went down the rabbit hole of creating an actual PipelineNode object with fake ids and I didn't like that at all. I also don't want the core type to be `Node | NewNode`, as only the configuration page can have a non-existing Node. Ideas?